ipa-core/src/protocol/ipa_prf/mod.rs

```diff
@@ -292,18 +298,30 @@ where
 
     let prf_key = gen_prf_key(&eval_ctx);
 
+    // CONV_CHUNK * PROOF_CHUNK
+    let validator = &convert_ctx.dzkp_validator(PROOF_CHUNK * CONV_CHUNK);
+
     let curve_pts = seq_join(
         ctx.active_work(),
-        process_slice_by_chunks(input_rows, move |idx, records: ChunkData<_, CONV_CHUNK>| {
-            let record_id = RecordId::from(idx);
-            let convert_ctx = convert_ctx.clone();
-            let input_match_keys: &dyn Fn(usize) -> Replicated<MatchKey> =
-                &|i| records[i].match_key.clone();
-            let match_keys =
-                BitDecomposed::<Replicated<Boolean, 256>>::transposed_from(input_match_keys)
-                    .unwrap_infallible();
-            convert_to_fp25519::<_, CONV_CHUNK, PRF_CHUNK>(convert_ctx, record_id, match_keys)
-        }),
+        process_slice_by_chunks(
+            input_rows,
+            move |idx, records: ChunkData<_, { PROOF_CHUNK * CONV_CHUNK }>| {
+                let vec_match_keys = (0usize..PROOF_CHUNK)
+                    .map(|j| {
+                        let input_match_keys: &dyn Fn(usize) -> Replicated<MatchKey> =
+                            &|i| records[CONV_CHUNK * j + i].match_key.clone();
+                        BitDecomposed::<Replicated<Boolean, 256>>::transposed_from(input_match_keys)
+                            .unwrap_infallible()
+                    })
+                    .collect::<Vec<_>>();
+                convert_to_fp25519::<_, C, CONV_CHUNK, PRF_CHUNK>(
+                    validator.clone(),
+                    idx,
+                    RecordId::from(idx * PROOF_CHUNK * CONV_CHUNK),
```

Comment:
I think this should be `idx * PROOF_CHUNK` -- the vectorized operation across CONV_CHUNK input rows only consumes one record ID. (But also see my other note about getting rid of the separate record ID parameter.)